test/storage/online_file_source.cpp

```diff
@@ -359,3 +359,56 @@ TEST(OnlineFileSource, TEST_REQUIRES_SERVER(NetworkStatusOnlineOffline)) {
 
     loop.run();
 }
+
+TEST(OnlineFileSource, TEST_REQUIRES_SERVER(RateLimit)) {
```

Comment:
We shouldn't add any more timing-dependent tests to the test suite. The ones we have already spuriously fail intermittently (as will most any timing-dependent test, eventually).

Instead, how about making `errorRetryTimeout` public and unit testing it instead?